bin/faucet/README.md

```diff
@@ -46,5 +46,151 @@ The faucet implements several security measures to prevent abuse:
+        let digest = BigInt('0x' + Array.from(hashBytes).map(b => b.toString(16).padStart(2, '0')).join(''));
+
+        // Check if the hash is less than the target
+        if (digest < target) {
+            console.log('Found nonce:', nonce);
+            return nonce;
+        }
+    } catch (error: any) {
+        throw new Error('Failed to compute hash: ' + error.message);
+    }
+}
+```
+
+3. **Request tokens** along with your solved challenge to `/get_tokens`
+
+```typescript
+const params = new URLSearchParams({
+    account_id: accountId,
+    is_private_note: 'true',
+    asset_amount: '100',
+    challenge: challenge,
+    nonce: nonce.toString()
+});
+
+const response = await fetch(`${baseUrl}/get_tokens?${params}`);
+if (!response.ok) throw new Error(`Get tokens error: ${response.status} ${await response.text()}`);
+
+const text = await response.text();
+const json = JSON.parse(text);
+const noteId = json.note_id;
+const txId = json.tx_id;
+const explorerUrl = json.explorer_url;
+```
+
+4. **Request note** to download generated private notes
+
+```typescript
+const response = await fetch(`${baseUrl}/get_note?note_id=${noteId}`);
+if (!response.ok) throw new Error(`Get note error: ${response.status} ${await response.text()}`);
+
+const text = await response.text();
+const json = JSON.parse(text);
+
+// Decode note with base64
+const noteData = Buffer.from(json.data_base64, 'base64');
+
+fs.writeFileSync('note.mno', noteData);
+```
+
+#### Example
+
+Check out the complete working examples below. Make sure the faucet is running at `http://localhost:8080` before using them.
+- Rust: [`examples/rust/request_tokens.rs`](examples/request_tokens.rs)
+- TypeScript: [`examples/typescript/request_tokens.ts`](examples/ts/request_tokens.ts)
+
+#### API Endpoints Reference
+
+**GET /pow**
+- **Purpose**: Request a proof-of-work challenge
+- **Query Parameters**:
+  - `account_id` (string, required): The account ID requesting the challenge
+  - `api_key` (string, optional): API key for authentication
+- **Response**: JSON object containing:
+  - `challenge` (string): The encoded challenge string in hexadecimal format
+  - `target` (number): The target value for the proof-of-work challenge. A solution is valid if the hash `H(challenge, nonce)` is less than this target. As the hashing function we use `SHA3-256`
+  - `timestamp` (number): The timestamp when the challenge was created (seconds since UNIX epoch)
+
+**GET /get_tokens**
+- **Purpose**: Request tokens
+- **Query Parameters**:
+  - `account_id` (string, required): The account ID requesting tokens
+  - `is_private_note` (boolean, required): Whether to create a private note
+  - `asset_amount` (number, required): Amount of tokens to request
+  - `challenge` (string, required): The encoded challenge from the `/pow` endpoint
+  - `nonce` (number, required): The nonce used to solve the challenge
+  - `api_key` (string, required): API key for authentication
```

Comment:
Is `api_key` required? I thought it would be optional, and if not submitted, and empty string would be assumed.